components/org.wso2.carbon.siddhi.extensions.installer.core/src/main/java/org/wso2/carbon/siddhi/extensions/installer/core/execution/SiddhiAppExtensionUsageDetectorImpl.java

```diff
@@ -0,0 +1,205 @@
+     * but are identifiable based on the regex of a unique attribute's value fall under this category.
+     * Eg:
+     * - type='cdc' (common).
+     * - url='jdbc:mysql://...' for MySQL.
+     * - url='jdbc:postgresql://...' for Postgres.
+     *
+     * @param usage Extension usage in a Siddhi app.
+     * @return Optional Extension config object if found.
+     */
+    private Optional<ExtensionConfig> matchByUniqueAttributeValueRegex(SiddhiAppExtensionUsage usage) {
+        Optional<Map.Entry<String, ExtensionConfig>> matchedExtensionEntry =
+            extensionConfigs.entrySet().stream()
+                .filter(extension -> {
+                    ExtensionIdentifierConfig extensionIdentifier = extension.getValue().getIdentifier();
+                    if (extensionIdentifier != null && extensionIdentifier.isUniqueAttributeValueRegexValid()) {
+                        String uniqueAttribute = extensionIdentifier.getUniqueAttribute();
+                        String uniqueAttributeValueRegex = extensionIdentifier.getUniqueAttributeValueRegex();
+                        return (
+                            Objects.equals(usage.getProperties().get(TYPE_KEY), extensionIdentifier.getType()) &&
+                                usage.getProperties().get(uniqueAttribute) != null &&
+                                usage.getProperties().get(uniqueAttribute).matches(uniqueAttributeValueRegex));
+                    }
+                    return false;
+                })
+                .findFirst();
+        return matchedExtensionEntry.map(Map.Entry::getValue);
+    }
+
+    /**
+     * Returns an optional extension config object,
+     * when the given usage matches an extension by an alternative type, as specified in the extension's
+     * {@link ExtensionIdentifierConfig}.
+     * Extensions such that, the same extension is denoted by different values for 'type' in Siddhi
+     * fall under this category.
+     * Eg:
+     * All of the following denote the 'http' extension.
+     * - type='http'.
+     * - type='http-service-response'.
+     * - type='http-call-response'.
+     *
+     * @param usage Extension usage in a Siddhi app.
+     * @return Optional Extension config object if found.
+     */
+    private Optional<ExtensionConfig> matchByAlternativeType(SiddhiAppExtensionUsage usage) {
+        Optional<Map.Entry<String, ExtensionConfig>> matchedExtensionEntry =
+            extensionConfigs.entrySet().stream()
+                .filter(extension -> {
+                    ExtensionIdentifierConfig extensionIdentifier = extension.getValue().getIdentifier();
+                    if (extensionIdentifier != null && extensionIdentifier.isAlternativeTypeValid()) {
+                        return extensionIdentifier.getAlternativeTypes().contains(usage.getProperties().get(TYPE_KEY));
+                    }
+                    return false;
+                })
+                .findFirst();
```

Comment:
shall we move this to 181

Reply:
Updated in https://github.com/wso2/carbon-analytics/pull/1840/commits/687d9d726cb4be51cef81b489166d694c78b91dd